server/src/main/java/org/elasticsearch/index/snapshots/IndexShardSnapshotStatus.java

```diff
@@ -50,6 +52,14 @@ public enum Stage {
          * Snapshot failed
          */
         FAILURE,
+        /**
+         * Snapshot pausing because of node removal
+         */
+        PAUSING,
+        /**
+         * Snapshot paused because of node removal
+         */
+        PAUSED,
```

Comment:
With these new stages, I wonder whether we should remove `AbortStatus` and replace it with just `Boolean`. I think that would make the code a bit simpler and one less crossed usages of abort and pause.

Reply:
I'm deliberately using an enum here because `true` and `false` are terrible names for the two states. Does `true` mean it aborted, or that it succeeded? Will you remember the answer when reading the code next time?

There is a subtle distinction between "abort" in the sense that the shard snapshot stopped before completion, possibly to resume later, and "abort" in the sense that the whole snapshot was aborted.